scapy/sendrecv.py

```diff
@@ -885,24 +890,39 @@ def sniff(count=0, store=True, offline=None, prn=None, lfilter=None,
         stoptime = time.time() + timeout
     remain = None
     read_allowed_exceptions = ()
-    is_python_can_socket = getattr(opened_socket, "is_python_can_socket", lambda: False)  # noqa: E501
+
+    are_python_can_sockets = lambda: \
+        all(list(filter(lambda sock:
+                        getattr(sock, "is_python_can_socket", lambda: False)(),
+                        sniff_sockets)))
```

Comment:
The `all` operator is powerful.
You can even use

```
all(s for s in sniff_sockets if hasattr(s, "is_python_can_socket"))
```

Reply:
well, this crashes my unit-tests. I'll work on fixing this.

Reply:
Of you are right, it should be 
```
all(s for s in sniff_sockets if getattr(s, "is_python_can_socket", lambda: False)())
```

Reply:
Hi @gpotter2, I get failing tests because the all operator. Your last proposal will do the following in my code: 
If a socket is a python_can socket, it will be in the temporary list, which `all` will get. In this case, I end up with `are_python_can_sockets = True`, which is correct. 
If a socket is not a python_can socket, it will create an empty temporary list. `all` will interpret this empty list, and will return `True`, by definition. In this case, I'll end up having `are_python_can_sockets = True`, which is wrong.

Reply:
I was saying this as an example
```
all(getattr(s, "is_python_can_socket", lambda: False)() for s in sniff_sockets)
```

Reply:
Ah ok, Thanks a lot.